source/Applications/Advanced/TransformViaArucoMarker/TransformViaArucoMarker.cpp

```diff
@@ -0,0 +1,308 @@
+
+        return markerPoints3D;
+    }
+
+    Zivid::Matrix4x4 transformationMatrix(const cv::Matx33d &rotationMatrix, const cv::Vec3d &translationVector)
+    {
+        auto transformMatrix = Zivid::Matrix4x4{ 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1 };
+
+        for(size_t row = 0; row < rotationMatrix.rows; row++)
+        {
+            for(size_t col = 0; col < rotationMatrix.cols; col++)
+            {
+                transformMatrix(row, col) = rotationMatrix(row, col);
+            }
+        }
+
+        transformMatrix(0, 3) = translationVector[0];
+        transformMatrix(1, 3) = translationVector[1];
+        transformMatrix(2, 3) = translationVector[2];
+
+        return transformMatrix;
+    }
+
+    void displayMarkerPose(const Zivid::PointCloud &pointCloud,
+                           const std::vector<cv::Point2f> &corners2D,
+                           const std::vector<cv::Point2f> &center2D)
+    {
+        auto rgbaImage = cv::Mat(pointCloud.height(), pointCloud.width(), CV_8UC4); // NOLINT(hicpp-signed-bitwise)
+        auto *ptr =
+            reinterpret_cast<Zivid::ColorRGBA *>(rgbaImage.data); // NOLINT(cppcoreguidelines-pro-type-reinterpret-cast)
+        pointCloud.copyData(ptr);
+        auto bgr = cv::Mat(pointCloud.height(), pointCloud.width(), CV_8UC4); // NOLINT(hicpp-signed-bitwise)
+        cv::cvtColor(rgbaImage, bgr, cv::COLOR_BGR2RGB);
+
+        auto poseCenter = center2D[0];
+        auto xAxisPoint = cv::Point2f(((corners2D[3].x + corners2D[0].x) / 2), ((corners2D[3].y + corners2D[0].y) / 2));
+        auto yAxisPoint = cv::Point2f(((corners2D[2].x + corners2D[3].x) / 2), ((corners2D[2].y + corners2D[3].y) / 2));
+
+        cv::circle(bgr, poseCenter, 2, cv::Scalar(255, 0, 0), 2);
+        cv::arrowedLine(bgr, poseCenter, xAxisPoint, cv::Scalar(0, 0, 255), 2);
+        cv::arrowedLine(bgr, poseCenter, yAxisPoint, cv::Scalar(0, 255, 0), 2);
+
+        cv::imshow("Aruco marker pose", bgr);
+        cv::waitKey(0);
+    }
+
+    Zivid::Matrix4x4 getMarkerPose(const Zivid::PointCloud &pointCloud,
+                                   const aruconano::Marker &markers,
+                                   const bool display = false)
+    {
+        const auto corners2D = get2DMarkerCorners(markers);
+        const auto centers2D = estimate2DMarkerCenters(markers);
+
+        const auto corners3D = estimate3DMarkerPoints(pointCloud, corners2D);
+        const auto centers3D = estimate3DMarkerPoints(pointCloud, centers2D);
+
+        auto origin = cv::Vec3d{ centers3D[0].x, centers3D[0].y, centers3D[0].z };
+        auto yAxis = cv::Vec3d{ corners3D[0].x - corners3D[1].x,
+                                corners3D[0].y - corners3D[1].y,
+                                corners3D[0].z - corners3D[1].z };
+
+        auto xAxis = cv::Vec3d{ corners3D[2].x - corners3D[1].x,
+                                corners3D[2].y - corners3D[1].y,
+                                corners3D[2].z - corners3D[1].z };
```

Comment:
Why yAxis first?

Reply:
no reason, and it the order does not matter. I can switch the order as its more common to do x first and then y